game-app/game-core/src/main/java/games/strategy/triplea/ui/SimpleUnitPanel.java

```diff
@@ -175,7 +89,7 @@ public void setUnitsFromRepairRuleMap(
     final Set<Unit> entries = units.keySet();
     for (final Unit unit : entries) {
       final IntegerMap<RepairRule> rules = units.get(unit);
-      final TreeSet<RepairRule> repairRules = new TreeSet<>(repairRuleComparator);
+      final TreeSet<RepairRule> repairRules = new TreeSet<>(RepairRule.getComparator());
```

Comment:
I like your solution to the problem, but I think it's better to have the abstract `Rule` class to implement `Comparable<Rule>` instead of providing a static method, then you can just omit this argument entirely.

Reply:
If the superclass works with compatibility there's no reason to change your code, it's pretty good already.
Just make sure to have a look at my newer comment about implementing the Comparable interface directly

Reply:
This is just about making the method non-static

Reply:
Thanks. It looks much better. 😁